src/equations/svaerd_kalisch_1d.jl

```diff
@@ -241,10 +241,10 @@ function rhs!(du_ode, u_ode, t, mesh, equations::SvaerdKalischEquations1D,
 
     # split form
     @timeit timer() "dv hyperbolic" begin
-        dv[:] = -(0.5 * (D1_central * (hv .* v) + hv .* D1v - v .* (D1_central * hv)) +
-                  equations.gravity * h .* D1eta +
-                  0.5 * (vD1y - D1vy - yD1v) -
-                  0.5 * D1_central * (gamma_hat .* (solver.D2 * v)) -
+        dv[:] = -(0.5 * (D1_central * (hv .* v) .+ hv .* D1v .- v .* (D1_central * hv)) .+
+                  equations.gravity * h .* D1eta .+
+                  0.5 * (vD1y .- D1vy .- yD1v) .-
+                  0.5 * D1_central * (gamma_hat .* (solver.D2 * v)) .-
                   0.5 * solver.D2 * (gamma_hat .* D1v))
```

Comment:
If you want to reduce memory allocations further, you could use something like
```suggestion
        D1_hv = D1_central * hv
        D1_hv2 = D1_central * (hv .* v)
        D1_gamma_hat_D2_v = D1_central * (gamma_hat .* (solver.D2 * v))
        D2_gamma_hat_D1_v = solver.D2 * (gamma_hat .* D1v)
        @. dv = -(0.5 * (D1_hv2 + hv * D1v - v * D1_hv) +
                  equations.gravity * h * D1eta +
                  0.5 * (vD1y - D1vy - yD1v) -
                  0.5 * D1_central * D1_gamma_hat_D2_v -
                  0.5 * D2_gamma_hat_D1_v)
```

Reply:
Thanks (I've changed the suggestion a bit because there was an additional `D1_central`. I will check the runtime locally.

Reply:
It actually helps a lot! Nice:

```julia
──────────────────────────────────────────────────────────────────────────────
        DispersiveSWE                 Time                    Allocations      
                             ───────────────────────   ────────────────────────
      Tot / % measured:           680ms /  98.6%           2.01GiB /  99.7%    

 Section             ncalls     time    %tot     avg     alloc    %tot      avg
 ──────────────────────────────────────────────────────────────────────────────
 rhs!                 5.16k    657ms   98.0%   127μs   1.99GiB   99.5%   405KiB
   dv elliptic        5.16k    556ms   82.9%   108μs   1.59GiB   79.2%   322KiB
   deta hyperbolic    5.16k   56.1ms    8.4%  10.9μs    250MiB   12.2%  49.5KiB
   dv hyperbolic      5.16k   40.6ms    6.1%  7.87μs    166MiB    8.1%  33.0KiB
   ~rhs!~             5.16k   4.30ms    0.6%   833ns    810KiB    0.0%     161B
   source terms       5.16k    244μs    0.0%  47.3ns     0.00B    0.0%    0.00B
 analyze solution        86   13.5ms    2.0%   157μs   10.0MiB    0.5%   119KiB
 ──────────────────────────────────────────────────────────────────────────────
```